lib/inspec/profile.rb

```diff
@@ -211,6 +212,44 @@ def locked_dependencies
       @locked_dependencies ||= load_dependencies
     end
 
+    def lockfile_exists?
```

Comment:
It really isn't clear to me how `lockfile_exists?()` is a side-effect?  Whether a lockfile is inside the profile is a property of the profile and depends on information (the path to the profile) that you don't necessarily want to expose to users of the class.